### PR TITLE
Metagame Multiplier and Limited-lock NFTs

### PR DESCRIPTION
**Pull request by figs999**

### CODE REVIEW

File: SuperChamps/interfaces/ISuperChampsToken.sol [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: None
+// Super Champs Foundation 2024
+
+pragma solidity ^0.8.24;
+
+import "./IPermissionsManager.sol";
+
+/// @title Interface which allows contracts to reference lock state and permissions manager from the CHAMP token.
+/// @author Chance Santana-Wees (Coelacanth/Coel.eth)
+interface ISuperChampsToken {
```
Comment (miteshJR) [inline SuperChamps/interfaces/ISuperChampsToken.sol:10]: @figs999 what is the use of this interface as superchampstoken is already minted out and it does not implement this interface in the defination

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,5 @@
 .deps
-contracts/SuperChamps/artifacts/*
-contracts/synthetix/contracts/artifacts/*
-contracts/synthetix/contracts/interfaces/artifacts/*
-*.json
```
Comment (miteshJR) [inline .gitignore:5]: *.json should be git ignored?

File: SuperChamps/contracts/Metagame/SCMetagameHouseRewards.sol [hunk 3]
```diff
@@ -50,10 +53,59 @@ contract SCMetagameHouseRewards is StakingRewards {
         require(access_pass.isVerified(msg.sender), "MUST HAVE VERIFIED ACCESS PASS");
         require(amount_ > 0, "CANNOT STAKE 0");
 
-        _totalSupply = _totalSupply + amount_;
+        uint256 _old_multiplier_basis_points = _multiplier_basis_points[msg.sender];
+        _multiplier_basis_points[msg.sender] = access_pass.getMetagameMultiplier(msg.sender);
+
+        _totalSupply -= _balances[msg.sender] * _old_multiplier_basis_points / 10000;
```
Comment (miteshJR) [inline SuperChamps/contracts/Metagame/SCMetagameHouseRewards.sol:59]: @figs999 whats the logic for division by 10000 everywhere?

File: SuperChamps/contracts/NFTs/SCTempLockedNFT.sol [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import "./SCDefaultRenderer.sol";
+import "../../interfaces/IPermissionsManager.sol";
+import "../../interfaces/ISuperChampsToken.sol";
+import "../../interfaces/IERC721MetadataRenderer.sol";
+
+/// @title Super Champs (CHAMP) Temporarily Transfer Locked NFT
+/// @author Chance Santana-Wees (Coelacanth/Coel.eth)
+/// @dev Token transfers are restricted to addresses that have the Transer Admin permission until the CHAMP token is unlocked.
+/// @notice This is a standard ERC721 token contract that restricts token transfers before the protocol token is unlocked. Trades still possible via Shop/Marketplace system, but limited to sales denominated in CHAMP.
+contract SuperChampsToken is ERC721 {
```
Comment (miteshJR) [inline SuperChamps/contracts/NFTs/SCTempLockedNFT.sol:17]: name confusing with SuperChampsToken rename to a bettername maybe

File: SuperChamps/contracts/Token/SuperChampsToken.sol [hunk 5]
```diff
@@ -25,13 +25,15 @@ contract SuperChampsToken is ERC20, ERC20Permit {
     modifier isAdminOrUnlocked() {
         require(!transfersLocked || 
                 permissions.hasRole(IPermissionsManager.Role.TRANSFER_ADMIN, _msgSender()) ||
-                permissions.hasRole(IPermissionsManager.Role.TRANSFER_ADMIN, tx.origin));
+                permissions.hasRole(IPermissionsManager.Role.TRANSFER_ADMIN, tx.origin),
```
Comment (miteshJR) [inline SuperChamps/contracts/Token/SuperChampsToken.sol:28]: will these changes have ant effect as token contract is already deployed?

File: SuperChamps/contracts/Utils/SCDeploymentHelper.sol [hunk 6]
```diff
@@ -23,12 +23,12 @@ contract SCDeploymentHelper {
     uint256 private constant TOTAL_SUPPLY = 1_000_000_000 ether;
 
     ///@notice The address of the Base multisig wallet that is controlled by Joyride
-    address private constant JOYRIDE = address(0xa8CAc43b28A7e5F0Ee797741195A920E88B8e7EB);
+    address private constant JOYRIDE = address(0x5B38Da6a701c568545dCfcB03FcB875f56beddC4);
```
Comment (miteshJR) [inline SuperChamps/contracts/Utils/SCDeploymentHelper.sol:26]: seems addresses used for testing are pushed?

File: SuperChamps/contracts/Utils/SCDeploymentHelper.sol [hunk 7]
```diff
@@ -23,12 +23,12 @@ contract SCDeploymentHelper {
...
     ///@notice The quantity of tokens to mint into the Base multisig wallet that is controlled by Joyride as a retained right.
     uint256 private constant JOYRIDE_ALLOCATION = 320_000_000 ether;
 
     ///@notice The address of the Base multisig wallet that is controlled by The Super Champs Foundation
-    address private constant SUPER_CHAMPS_FOUNDATION = address(0x423bA9539440ec39B573Ed5F0eD98e1Ed39cD634);//address(0xAb8483F64d9C6d1EcF9b849Ae677dD3315835cb2); //need correct address
+    address private constant SUPER_CHAMPS_FOUNDATION = address(0xAb8483F64d9C6d1EcF9b849Ae677dD3315835cb2);//address(0xAb8483F64d9C6d1EcF9b849Ae677dD3315835cb2); //need correct address
```
Comment (miteshJR) [inline SuperChamps/contracts/Utils/SCDeploymentHelper.sol:31]: is this change needed or just did for testing?

File: SuperChamps/interfaces/ISuperChampsToken.sol (hunk 1 above)
Comment (figs999) [reply SuperChamps/interfaces/ISuperChampsToken.sol:10]: Interfaces do not actually need to be included in a contract for them to conform to them.
This interface was included just to be syntactic sugar that exposes specific methods without requiring the entire contract code to be compiled into future contracts.

File: .gitignore (hunk 2 above)
Comment (figs999) [reply .gitignore:5]: This could be made less broad if needed. But I don't want all of the build artifacts to be included in the repo.

File: SuperChamps/contracts/Metagame/SCMetagameHouseRewards.sol (hunk 3 above)
Comment (figs999) [reply SuperChamps/contracts/Metagame/SCMetagameHouseRewards.sol:59]: The multiplier is in basis points.

File: SuperChamps/contracts/NFTs/SCTempLockedNFT.sol (hunk 4 above)
Comment (figs999) [reply SuperChamps/contracts/NFTs/SCTempLockedNFT.sol:17]: This was a copy-pasta. You are correct it should be different.

File: SuperChamps/contracts/Token/SuperChampsToken.sol (hunk 5 above)
Comment (figs999) [reply SuperChamps/contracts/Token/SuperChampsToken.sol:28]: This minor change just never got merged in.

File: SuperChamps/contracts/Utils/SCDeploymentHelper.sol (hunk 7 above)
Comment (figs999) [reply SuperChamps/contracts/Utils/SCDeploymentHelper.sol:31]: Just for testing. we can revert this file.

**Review by miteshJR — APPROVED** [review]
